aya/src/programs/mod.rs

```diff
@@ -386,6 +386,18 @@ impl<T: Link> ProgramData<T> {
     }
 }
 
+fn unload_program<T: Link>(data: &mut ProgramData<T>, detach: bool) -> Result<(), ProgramError> {
+    if detach {
+        data.links.remove_all()?;
+    }
+    let fd = data.fd_or_err()?;
```

Comment:
This would be better:
```rust
if let Some(fd) = data.fd.take() {
   unsafe {
        libc::close(fd);
    }
} else {
   return Err(ProgramError::NotLoaded)
}
```